x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -68,15 +68,42 @@ export async function setupFleet(
   esClient: ElasticsearchClient
 ): Promise<SetupStatus> {
   const t = apm.startTransaction('fleet-setup', 'fleet');
+  const logger = appContextService.getLogger();
 
   try {
+    // check if fleet setup is already started
+    const settings = await settingsService.getSettingsOrUndefined(soClient);
+
+    if (settings && settings.fleet_setup_status === 'in_progress') {
+      logger.info('Fleet setup already in progress, aborting');
+      return {
+        isInitialized: false,
+        nonFatalErrors: [],
+      };
+    }
+
+    try {
+      await settingsService.saveSettings(soClient, {
+        fleet_setup_status: 'in_progress',
+      });
+    } catch (error) {
+      logger.warn(`Error setting fleet setup status to in_progress: ${error}`);
+    }
+
     return await awaitIfPending(async () => createSetupSideEffects(soClient, esClient));
   } catch (error) {
     apm.captureError(error);
     t.setOutcome('failure');
     throw error;
   } finally {
     t.end();
+    try {
+      await settingsService.saveSettings(soClient, {
+        fleet_setup_status: 'completed',
```

Comment:
Added writing out the `completed` status to the `finally` branch to make sure it is written out in case of any errors.
Leaving this flag `in_progress` would prevent any subsequent Fleet setup calls from running.

Reply:
Adding a force flag makes sense for edge cases and troubleshooting, imo